### PR TITLE
Fix map access in order

### PR DESCRIPTION
**Pull request by timowest**
Fixes https://github.com/querydsl/querydsl/issues/1624

### CODE REVIEW

File: querydsl-jpa/src/main/java/com/querydsl/jpa/JPAListAccessVisitor.java [hunk 1]
```diff
@@ -28,10 +28,13 @@
...
     private final QueryMetadata metadata;
 
+    private final Map<Expression<?>, Path<?>> aliases;
+
     private final Map<Path<?>, Path<?>> replacements = Maps.newHashMap();
 
-    public JPAListAccessVisitor(QueryMetadata metadata) {
+    public JPAListAccessVisitor(QueryMetadata metadata, Map<Expression<?>, Path<?>> aliases) {
         this.metadata = metadata;
+        this.aliases = aliases;
```
Comment (Shredder121) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/JPAListAccessVisitor.java]: Shouldn't these be used?
I don't see a method similar to `JPAMapAccessVisitor.shorten`.

Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/JPAListAccessVisitor.java]: I applied now the same shortening for list access

**Comment by Shredder121** [comment]
Could you also add a test case that hits [these lines](https://coveralls.io/builds/4008292/source?filename=querydsl-jpa%2Fsrc%2Fmain%2Fjava%2Fcom%2Fquerydsl%2Fjpa%2FJPAListAccessVisitor.java#L68)?

**Comment by timowest** [comment]
Test coverage should be improved now.

**Comment by Shredder121** [comment]
Looks good to me.

**Comment by johnktims** [comment]
Looks good.

**Pull request merged**